examples/debug/src/TestRun.hs

```diff
@@ -0,0 +1,147 @@
+import System.Environment (getArgs, getEnv, setEnv)
+import System.FilePath ((</>))
+import TimeDebugContract qualified
+import Tools
+import Wallet.Types (ContractInstanceId (ContractInstanceId))
+import Prelude
+
+main :: IO ()
+main = testnetRun
+
+testnetRun :: IO ()
+testnetRun = do
+  setLocaleEncoding utf8
+  [bpiDir, cliDir, sockPath, netMagic, operation] <- getArgs -- /home/mike/dev/mlabs/net-setups/testnet-bpi-setup/data
+  setEnv "CARDANO_NODE_SOCKET_PATH" sockPath
+  getEnv "PATH" >>= \p -> setEnv "PATH" (p ++ ":" ++ cliDir)
+
+  let netMagic' = read netMagic
+
+  cEnv <- mkContractEnv netMagic' bpiDir
+
+  putStrLn "Running contract"
+
+  -- putStrLn $ "=== Stats ===\n" ++ show stats
+  void $ runMyContract cEnv operation
+  collateral <- readTVarIO . unCollateralVar . ceCollateral $ cEnv
+  -- stats <- readTVarIO (ceContractStats cEnv)
+  putStrLn $ "Collateral env: " <> show collateral
+  where
+    runMyContract cEnv operation = do
+      res <- case operation of
+        "light" -> do
+          putStrLn "Running loght debug"
+          BPI.runContract cEnv TimeDebugContract.timeDebugLight
+        "split" -> do
+          putStrLn "Splitting whatever first utxo"
+          fmap show <$> BPI.runContract cEnv TimeDebugContract.splitUtxo
+        "lock" -> do
+          putStrLn "Locking"
+          BPI.runContract cEnv TimeDebugContract.lockAtScript
+        "unlock" -> do
+          putStrLn "Spending"
+          BPI.runContract cEnv TimeDebugContract.unlockWithTimeCheck
+        "lock-unlock" -> do
+          putStrLn "Lock -> Unlock"
+          BPI.runContract cEnv TimeDebugContract.lockUnlock
+        "viaPay" -> do
+          putStrLn "Debug range validation and mempool. Specify interval:"
+          int <- read <$> getLine
+          BPI.runContract cEnv (TimeDebugContract.timeDebugViaPay int)
+        "collateral" -> do
+          putStrLn "payToHardcodedPKH"
+          BPI.runContract cEnv SomeDebugContract.payToHardcodedPKH >> pure (Right "Done")
+        other -> error $ "Unsupported operation: " ++ other
+
+      case res of
+        -- Right r -> "=== OK ===\n" ++ show r >> runMyContract
+        Right r -> do
+          putStrLn ("=== OK ===\n" ++ show r)
+        -- randomDelay
+        -- runMyContract cEnv operation
+        Left e -> putStrLn ("=== FAILED ===\n" ++ show e) -- >> return (show e)
+
+type NetMagic = Integer -- 0 fot mainnet, 1097911063 public testnet
+
+mkContractEnv :: Monoid w => NetMagic -> FilePath -> IO (ContractEnvironment w)
+mkContractEnv netMagic bpiDir = do
+  let paramsFile = bpiDir </> "pparams.json"
+  Just pparams <- decodeFileStrict paramsFile
+  contractInstanceID <- ContractInstanceId <$> UUID.nextRandom
+  contractState <- newTVarIO (ContractState Active mempty)
+  contractStats <- newTVarIO (ContractStats mempty)
+  contractLogs <- newTVarIO (LogsList mempty)
+  collateral <- CollateralVar <$> newTVarIO Nothing
+  pkhs <- getPkhs bpiDir
+  return $
```

Comment:
This applies to this line and a a few other changes in this PR: most of the codebase uses `pure`, so I think that should be kept consistent

Reply:
Fixed